cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -44,6 +44,112 @@ namespace training
 {
 namespace internal
 {
+//////////////////////////////////////////////////////////////////////////////////////////
+// Service class, it uses to keep information about nodes
+//////////////////////////////////////////////////////////////////////////////////////////
+template <typename WorkItem>
+class BinaryHeap
```

Comment:
1. [Here ](https://github.com/oneapi-src/oneDAL/blob/master/cpp/daal/src/algorithms/service_heap.h) implemented as far as I know a binary heap. Are there any reasons to use your own implementation?
2. If you use self implementation, you will move here to common place, see example cpp/daal/src/algorithms/service_heap.h
3. I see this class not contain `cpu` specific -> this increases the risks of getting `illegal instruction`

Reply:
thanks, it was fixed